go/vt/vtctl/workflow/server.go

```diff
@@ -2537,6 +2574,19 @@ func (s *Server) switchWrites(ctx context.Context, req *vtctldatapb.WorkflowSwit
 		ts.Logger().Errorf("createJournals failed: %v", err)
 		return 0, nil, err
 	}
+	// Initialize any target sequences, if there are any, before allowing new writes.
+	if req.InitializeTargetSequences && len(sequenceMetadata) > 0 {
+		// Writes are blocked so we can safely initialize the sequence tables but
+		// we also want to use a shorter timeout than the parent context.
+		// We use up at most half of the overall timeout.
+		initSeqCtx, cancel := context.WithTimeout(ctx, timeout/2)
+		defer cancel()
+		if err := sw.initializeTargetSequences(initSeqCtx, sequenceMetadata); err != nil {
+			werr := vterrors.Wrapf(err, "initializeTargetSequences failed")
+			ts.Logger().Error(werr)
```

Comment:
How about a new inner function, like
```go
logError := func(err error) error {
  ts.Logger().Error(err)
  return err
}
```

then using it like so:
```go
return 0, nil, logError(vterrors.Wrapf(err, "initializeTargetSequences failed"))
```

I'm suggesting this because I see in this function many places that repeat the above logic, _incorrectly_. For example, a few lines below we have:

```go
		ts.Logger().Errorf("allowTargetWrites failed: %v", err)
		return 0, nil, err
```
Which is a shame because the error logged and the error returned are not identical, and can make for more complex debugging. So I think we can&should replace all similar error paths in this function like so. I realize this comment is tangential to this PR.

Reply:
I disagree that it is *incorrect*. One is a log message -- targeted at administrators -- and the other an error that is shown to the user. It is perfectly fine and normal to give additional information to the administrator that is not presented to the end user. The user of the system doesn't really care what internal vitess function was involved -- whereas the administrator, if they are working on the codebase does. I don't agree that this makes things harder to debug either. This is pretty standard in software (take Vitess and MySQL for example). 

I can remove `allowTargetWrites failed:` portion in the log, but I disagree with the reasoning.

Reply:
No strong opinion. At your discretion.